### ISSUE TITLE
http and https apache setups are not consistent

### ISSUE THREAD

**Issue by nblumoe** [l3iggs/docker-owncloud#3]
Here is the relevant discussion from Docker Hub so far (youngest to oldest):

l3iggs

@blumoe do you get the 404 errors if you don't try to mount your own config and data files? just run it as:

docker run –name oc -p 80:80 -p 443:443 -d l3iggs/owncloud

and see if https works for you

nblumoe:
The run command looks like this:

docker run -v ~/ocfiles:/usr/share/webapps/owncloud/data -v ~/occonfig:/etc/webapps/owncloud/config –name oc2 -p 80:80 -p 443:443 -d l3iggs/owncloud

I am not using my own certificates (yet) but the ones included in your image. The same problem occurs with recent firefox and chrome version on Ubuntu (I am pretty sure it's not a browser issue).

What came to my mind is, that this is not an issue with the docker container, but rather the host OS. I am running this on a digital ocean machine, which might not have port 443 properly setup, will check this now.

l3iggs:
Hi @nblumoe could you provide a few more details? What options exactly are you running docker with? Did you try to generate your own ssl certificates or are you using the ones I already mad in the image? Have you tried a different browser?

nblumoe:
This went pretty smooth so far, thanks for the image! However, currently https is not working for me. As expected I got the warning about the untrusted certificate and I added an exception for it. But then I end up with 404 errors for all the ownloud url paths and a blank index page for the root path. Seems to be an apache config issue.

**Comment by nblumoe** [comment]
@l3iggs good idea to move it over here.

Running just `docker run --name oc -p 80:80 -p 443:443 -d l3iggs/owncloud` results in exactly the same behaviour.

**Comment by l3iggs** [comment]
Hmmm. I can't recreate this. When exactly do you get the 404 errors? Do you get the error when you visit https://localhost/owncloud immediately after starting the image?

Do you have the latest image (`docker pull l3iggs/owncloud`)?

**Comment by nblumoe** [comment]
Okay, I could reproduce it on my local machine and here is the issue:

http://localhost -> works
https://localhost -> doesn't work
https://localhost/owncloud -> works

So, when using https it depends on the path `/owncloud` being used, whereas for simple http it doesn't. Can you make this behave consistently by changing one or the other?

**Comment by l3iggs** [comment]
Thanks for pointing this out @nblumoe
I've addressed this with 28519404516fbb231e449bd00a3ea6c40b00c56f and d11da0b59e36c2179252baf66b77c8a6480645c5. Attempt to access the server root via http and https should now give an access forbidden message (error 403).

I'd say it's a fairly significant apache config change, so please test it and let me know if anything new is broken and get back to me here if you can.

The new image should be ready to pull from the docker registry in an hour or so.

**Comment by nblumoe** [comment]
Is it possible to do a non-breaking upgrade of an existing container / owncloud installation to the new image version?

**Comment by l3iggs** [comment]
Yes. Is possible. I migrate between docker images without losing anything important. I'm very careful the keep everything that I want to be persistent stored outside of the container though (that means data files, config files (as described in the readme) and the OC database).

It sounds like you've been keeping your data files and your config files outside of the image so I don't expect any problems there. You might have issues with your database (depending on how you configured it during setup). I haven't written up anything about that in the readme. When I first went through the owncloud setup, I pointed owncloud at the mysql server I have running on my host machine (not the one running in the docker image as it defaults to). 

I don't feel comfortable promising you that you won't have any problems switching containers without data loss.
